examples/secp256k1-usdt/src/lib.ts

```diff
@@ -0,0 +1,268 @@
+    }
+  }
+
+  for await (const cell of indexer
+    .collector({ lock: helpers.parseAddress(holderAccountInfo.address, { config: AGGRON4 }), type: "empty" })
+    .collect()) {
+    holderCKBCells.push(cell);
+  }
+
+  txSkeleton = txSkeleton.update("inputs", (inputs) => {
+    let total = BI.from(0);
+    for (const cell of issuerSUDTCells) {
+      if (total.gt(BI.from(CKBAmount).mul(CKB2SUDTRate))) {
+        break;
+      }
+      inputs = inputs.push(cell);
+      total = total.add(readBigUInt128LECompatible(cell.data));
+    }
+    return inputs;
+  });
+
+  const issuerTypeScript = {
+    code_hash: SUDT_SCRIPT.CODE_HASH,
+    hash_type: SUDT_SCRIPT.HASH_TYPE,
+    args: computeScriptHash(issuerAccountInfo.lockScript),
+  };
+
+  const SUDTTargetOutput: Cell = {
+    cell_output: {
+      capacity: "0x0",
+      lock: holderAccountInfo.lockScript,
+      type: issuerTypeScript,
+    },
+    data: toBigUInt128LE(BI.from(CKBAmount).mul(CKB2SUDTRate)),
+  };
+
+  const SUDTChangeOutput: Cell = {
+    cell_output: {
+      capacity: "0x0",
+      lock: issuerAccountInfo.lockScript,
+      type: issuerTypeScript,
+    },
+    data: toBigUInt128LE(calculateSUDTBalance(issuerSUDTCells).sub(BI.from(CKBAmount).div(1e8).mul(CKB2SUDTRate))),
+  };
+
+  SUDTTargetOutput.cell_output.capacity = BI.from(helpers.minimalCellCapacity(SUDTTargetOutput)).toHexString();
+  SUDTChangeOutput.cell_output.capacity = BI.from(helpers.minimalCellCapacity(SUDTChangeOutput)).toHexString();
+
+  txSkeleton = txSkeleton.update("outputs", (outputs) => {
+    return outputs.push(SUDTTargetOutput, SUDTChangeOutput);
+  });
+
+  const holderCKBInputCells: Cell[] = [];
+
+  let amount = BI.from(0);
+  for (const cell of holderCKBCells) {
+    if (amount.gt(CKBAmount)) {
+      break;
+    }
+    holderCKBInputCells.push(cell);
+    amount = amount.add(cell.cell_output.capacity);
+  }
+  txSkeleton = txSkeleton.update("inputs", (inputs) => {
+    return inputs.push(...holderCKBInputCells);
+  });
+
+  const CKBOutput: Cell = {
+    cell_output: {
+      capacity: BI.from(CKBAmount).toHexString(),
+      lock: issuerAccountInfo.lockScript,
+    },
+    data: "0x",
+  };
+
+  // exchanged CKB from holder to issuer and change
+  txSkeleton = txSkeleton.update("outputs", (outputs) => {
+    return outputs.push(CKBOutput);
+  });
+
+  const inputsCapacity = txSkeleton.inputs.reduce((acc, cur) => acc.add(cur.cell_output.capacity), BI.from(0));
+  const outputsCapacity = txSkeleton.outputs.reduce((acc, cur) => acc.add(cur.cell_output.capacity), BI.from(0));
+
+  const changeCapacity = inputsCapacity.sub(outputsCapacity);
+
+  const CKBChangeOutput: Cell = {
+    cell_output: {
+      capacity: changeCapacity.toHexString(),
+      lock: holderAccountInfo.lockScript,
+    },
+    data: "0x",
+  };
+  txSkeleton = txSkeleton.update("outputs", (outputs) => outputs.push(CKBChangeOutput));
+
+  txSkeleton = txSkeleton.update("witnesses", (witnesses) => {
+    const dummyWitness = {
```

Comment:
```suggestion
    const placeholderWitness = {
```

we should avoid creating new concept

Reply:
recommend using the `prepareSigningEntries` to avoid filling placeholder witness manually